gateway/src/main/java/org/georchestra/gateway/filter/headers/ServiceErrorGatewayFilterFactory.java

```diff
@@ -0,0 +1,46 @@
+package org.georchestra.gateway.filter.headers;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import org.georchestra.gateway.filter.global.ResolveTargetGlobalFilter;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.GatewayFilterChain;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.core.Ordered;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.server.reactive.ServerHttpResponse;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+import javax.validation.constraints.NotEmpty;
+
+public class ServiceErrorGatewayFilterFactory extends AbstractGatewayFilterFactory<Object> {
+    public ServiceErrorGatewayFilterFactory() {
+        super(Object.class);
+    }
+
+    @Override
+    public GatewayFilter apply(final Object config) {
+        return new ServiceErrorGatewayFilter();
+    }
+
+    private static class ServiceErrorGatewayFilter implements GatewayFilter, Ordered {
+
+        public @Override Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+            return chain.filter(exchange).then(Mono.fromRunnable(() -> {
+                if (exchange.getResponse().getStatusCode().is4xxClientError() || exchange.getResponse().getStatusCode().is5xxServerError()) {
+                    // log error ?
+                    throw new RuntimeException("SERVICE ERROR");
```

Comment:
It'd be ideal if we throw an exception that preserves the status code. I can't remember how they're called right now though

Reply:
`org.springframework.web.server.ResponseStatusException` that's it